### PR TITLE
`find-lib` refactor

### PR DESCRIPTION
**Pull request by robbiepitts**
Addresses #4.

Also adds tests.

### CODE REVIEW

File: test/find-lib-test.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+      expect(foundLib).to.eq(path.join(__dirname, '..', 'node_modules', 'foo', 'dist'));
+    });
+  });
+});
+
+function setupFakePackage(packageJson) {
+  fs.mkdirSync(path.join(__dirname, '../node_modules/foo'));
+  fs.writeFileSync(path.join(__dirname, '../node_modules/foo/package.json'), JSON.stringify(packageJson));
+}
```
Comment (robbiepitts) [inline test/find-lib-test.js]: This was the best way I could think of to test this. Please advise if there's a better way.

**Comment by dgeb** [comment]
@robbiepitts this is looking great 👍 

Could you just add one more test to verify that `module` is favored over `main` when both are defined?

**Comment by robbiepitts** [comment]
@dgeb Good call. Will do.

**Comment by robbiepitts** [comment]
@dgeb ✅

**Comment by dgeb** [comment]
Thanks @robbiepitts :shipit:

**Pull request merged**